File: backend/src/main/java/br/senai/saepveterinaria/entity/Usuario.java
package br.senai.saepveterinaria.entity;

import br.senai.saepveterinaria.enums.RoleUsuario;
import jakarta.persistence.*;
import jakarta.validation.constraints.Email;
import jakarta.validation.constraints.NotBlank;
import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Getter;
import lombok.NoArgsConstructor;
import lombok.Setter;

@Entity
@Table(
        name = "usuario",
        indexes = {
                @Index(name = "idx_usuario_status", columnList = "status_usuario")
        }
)
@Getter
@Setter
@AllArgsConstructor
@NoArgsConstructor
@Builder
public class Usuario {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Integer idUsuario;

    @Version
    private Long version;

    @NotBlank(message = "O nome é obrigatório")
    @Column(nullable = false)
    private String nome;

    @NotBlank(message = "O email é obrigatório")
    @Column(nullable = false, unique = true, length = 100)
    @Email
    private String email;

    @NotBlank(message = "A senha é obrigatório")
    @Column(nullable = false, length = 255)
    private String senha;

    @Enumerated(EnumType.STRING)
    @Column(nullable = false, length = 20)
    private RoleUsuario role;

    @Builder.Default
    @Column(nullable = false, columnDefinition = "boolean default true")
    private Boolean statusUsuario = true;
}

File: frontend/src/types/usuario.ts
export interface AlterarSenhaDTO {
    senhaAtual: string;
    novaSenha: string;
}

export interface UsuarioCreateDTO {
    nome: string;
    email: string;
    senha: string;
    role: string;
}

export interface UsuarioResponseDTO {
    idUsuario: number;
    nome: string;
    email: string;
    role: string;
    statusUsuario: boolean;
}

export interface UsuarioUpdateDTO {
    nome: string;
    email: string;
    role: string;
}
